### PR TITLE
Update Security documentation [8462]

### PR DESCRIPTION
**Pull request by rsanchez15**

### CODE REVIEW

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/376/)

File: docs/fastdds/security/access_control_plugin/access_control_plugin.rst [hunk 1]
```diff
@@ -0,0 +1,523 @@
...
+The DDS\:Access\:Permissions authentication plugin, can be activated setting the |DomainParticipantQos| |Property|
+``dds.sec.auth.plugin`` with the value ``builtin.Access-Permissions``.
+The following table outlines the properties used for the DDS\:Access\:Permissions plugin configuration.
+
+.. list-table:: **Properties to configure the DDS:Access:Permissions plugin**
+   :header-rows: 1
+   :align: left
+
+   * - Property name |br|
+       (all properties have |br| "dds.sec.access.builtin.Access-Permissions." |br| prefix)
```
Comment (EduPonz) [inline docs/fastdds/security/access_control_plugin/access_control_plugin.rst:49]: Same as the authentication table

File: docs/fastdds/security/access_control_plugin/access_control_plugin.rst [hunk 2]
```diff
@@ -0,0 +1,523 @@
...
+.. literalinclude:: /../code/GovernanceTester.xml
+   :language: xml
+   :start-after: <!-->GOVERNANCE_EXAMPLE<-->
+   :end-before: <!--><-->
+   :linenos:
+
+The `Governance XSD file <https://github.com/eProsima/Fast-RTPS/blob/master/resources/xsd/governance.xsd>`_ and
+the
+`Governance XML example <https://github.com/eProsima/Fast-RTPS/blob/master/examples/C%2B%2B/SecureHelloWorldExample/certs/governance.xml>`_
+can also be downloaded from the `*eProsima Fast DDS* Github repository <https://github.com/eProsima/Fast-RTPS>`_.
```
Comment (EduPonz) [inline docs/fastdds/security/access_control_plugin/access_control_plugin.rst:111]:
```suggestion
can also be downloaded from the `eProsima Fast DDS Github repository <https://github.com/eProsima/Fast-RTPS>`_.
```

File: docs/fastdds/security/access_control_plugin/access_control_plugin.rst [hunk 3]
```diff
@@ -0,0 +1,523 @@
...
+
+.. toctree::
+   :hidden:
+
+   governance.rst
+
+Domain Rules
+************
+
```
Comment (EduPonz) [inline docs/fastdds/security/access_control_plugin/access_control_plugin.rst:120]: I'd elaborate more on what rules are

File: docs/fastdds/security/access_control_plugin/access_control_plugin.rst [hunk 4]
```diff
@@ -0,0 +1,523 @@
...
+*  ``true``: the DomainParticipant shall check the permissions of the authenticated remote DomainParticipant.
+
+Discovery Protection Kind
+#########################
+
+This element is delimited by the ``<discovery_protection_kind>`` XML element tag.
+Indicates whether the secure channel of the endpoint discovery phase needs to be encrypted.
+The possible values are:
+
+*  ``SIGN`` or ``ENCRYPT``: the secure channel shall be encrypted.
```
Comment (EduPonz) [inline docs/fastdds/security/access_control_plugin/access_control_plugin.rst:228]: In general, is there any difference between `SIGN` and `ENCRYPT`?

File: docs/fastdds/security/access_control_plugin/access_control_plugin.rst [hunk 5]
```diff
@@ -0,0 +1,523 @@
...
+.. literalinclude:: /../code/PermissionsTester.xml
+   :language: xml
+   :start-after: <!-->PERMISSIONS_EXAMPLE<-->
+   :end-before: <!--><-->
+   :linenos:
+
+The `Permissions XSD file <https://github.com/eProsima/Fast-RTPS/blob/master/resources/xsd/governance.xsd>`_ and
+the
+`Permissions XML example <https://github.com/eProsima/Fast-RTPS/blob/master/examples/C%2B%2B/SecureHelloWorldExample/certs/governance.xml>`_
+can also be downloaded from the `*eProsima Fast DDS* Github repository <https://github.com/eProsima/Fast-RTPS>`_.
```
Comment (EduPonz) [inline docs/fastdds/security/access_control_plugin/access_control_plugin.rst:398]: In general:
```suggestion
can also be downloaded from the `eProsima Fast DDS Github repository <https://github.com/eProsima/Fast-RTPS>`_.
```

File: docs/fastdds/security/access_control_plugin/access_control_plugin.rst [hunk 6]
```diff
@@ -0,0 +1,523 @@
...
+   :end-before: <!--><-->
+
+Signing documents using x509 certificate
+""""""""""""""""""""""""""""""""""""""""
+
+:ref:`domain_governance_doc` and :ref:`domainparticipant_permissions_doc` have to be signed using an X.509 certificate.
+Generation of an X.509 certificate is explained in :ref:`generate_x509`.
+Next commands sign the necessary documents for its use by the DDS\:Access\:Permissions plugin.
+
+.. code-block:: bash
```
Comment (EduPonz) [inline docs/fastdds/security/access_control_plugin/access_control_plugin.rst:517]: I general, in Lunix

File: docs/fastdds/security/crypto_plugin/crypto_plugin.rst [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+| **Property name**                        | **Description**                                 | **Property Value**      |
++==========================================+=================================================+=========================+
+| rtps.participant.rtps_protection_kind    | Encrypt whole RTPS messages                     | ``ENCRYPT``             |
++------------------------------------------+-------------------------------------------------+-------------------------+
+| rtps.endpoint.submessage_protection_kind | Encrypt RTPS submessages of a particular entity | ``ENCRYPT``             |
++------------------------------------------+-------------------------------------------------+-------------------------+
+| rtps.endpoint.payload_protection_kind    | Encrypt payload of a particular Writer          | ``ENCRYPT``             |
++------------------------------------------+-------------------------------------------------+-------------------------+
+
+The following is an example of how to set the properties of |DomainParticipantQoS| for the DDS\:Crypto\:AES-GCM-GMAC
```
Comment (EduPonz) [inline docs/fastdds/security/crypto_plugin/crypto_plugin.rst:50]: Maybe one sub-section for each entity

File: docs/spelling_wordlist.txt [hunk 8]
```diff
@@ -1,3 +1,26 @@
...
+LogTopic
+Nagle
+PKI
+QoS
+Structs
+TCP
+Topologies
+TypeObject
+WString
+XTypes
```
Comment (EduPonz) [inline docs/spelling_wordlist.txt:23]: Careful with this

File: docs/fastdds/security/access_control_plugin/access_control_plugin.rst (hunk 6 above)
Comment (rsanchez15) [reply docs/fastdds/security/access_control_plugin/access_control_plugin.rst:517]: All openssl commands are the same for windows and linux.

File: docs/fastdds/security/crypto_plugin/crypto_plugin.rst (hunk 7 above)
Comment (rsanchez15) [reply docs/fastdds/security/crypto_plugin/crypto_plugin.rst:50]: The logging plugin only applies to DomainParticipants.

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/415/)

File: docs/fastdds/security/includes/intro.rst [hunk 9]
```diff
@@ -1,12 +1,47 @@
...
+In compliance with the `DDS Security <https://www.omg.org/spec/DDS-SECURITY/1.1/>`_ specification, Fast DDS provides
+secure communication by implementing pluggable security at three levels: a) |DomainParticipants| authentication
+(DDS\:Auth\:PKI-DH), b) access control of Entities (DDS\:Access\:Permissions), and c) data encryption
+(DDS\:Crypto\:AES-GCM-GMAC).
+Furthermore, for the monitoring of the security plugins and logging relevant events, Fast DDS implements
+the logging plugin (DDS\:Logging\:DDS_LogTopic).
+
+.. note::
+  Currently the  DDS\:Tagging\:DDS_Discovery plugin is not implemented in Fast DDS.
+  Its implementation is expected for future release of Fast DDS.
```
Comment (EduPonz) [inline docs/fastdds/security/includes/intro.rst:30]: I'd move this note to just below the list

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/418/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/420/)

**Pull request merged**